### PR TITLE
Fix/v33 remaining bugs

### PR DESCRIPTION
**Pull request by sedghi**
- warn user for overlapping segments which is not currently supported
- more general MPR which can be applied on any porotocol; not only on default HP like before

### CODE REVIEW

File: extensions/cornerstone/src/commandsModule.ts [hunk 1]
```diff
@@ -160,48 +146,51 @@ const commandsModule = ({ servicesManager }) => {
...
 
-        // Set the new toolName to be active
-        toolGroup.setToolActive(toolName, {
-          bindings: [{ mouseButton: Enums.MouseBindings.Primary }],
-        });
+      const activeToolName = toolGroup.getActivePrimaryMouseButtonTool();
 
-        return;
+      if (activeToolName) {
+        // Todo: this is a hack to prevent the crosshairs to stick around
```
Comment (wayfarer3130) [inline extensions/cornerstone/src/commandsModule.ts:178]: Discuss in slack

File: extensions/cornerstone/src/utils/mpr/toggleMPRHangingProtocol.ts [hunk 2]
```diff
@@ -0,0 +1,285 @@
+import { Enums } from '@cornerstonejs/tools';
```
Comment (wayfarer3130) [inline extensions/cornerstone/src/utils/mpr/toggleMPRHangingProtocol.ts:1]: Thank you for splitting this out - it makes it easier to understand to have things in smaller pieces.

**Review by wayfarer3130 — CHANGES_REQUESTED** [review]
Lets look at the set active/inactive and see if that code can be moved into the Tools instances themselves, rather than having the if condition on the two types.  Otherwise the changes look good.

**Comment by sedghi** [comment]
@wayfarer3130 Sure, as discussed privately, we will do it after the release

**Review by wayfarer3130 — APPROVED** [review]
Changes look good now - just make sure we have an issue to update the handling of tools with respect to whether they are active etc.

**Pull request merged**